src/Node/ForNode.php

```diff
@@ -72,13 +72,7 @@ public function compile(Compiler $compiler): void
         ;
 
         if ($this->hasNode('else')) {
-            $compiler
-                ->write("if (0 === \$iterator->getIndex0()) {\n")
-                ->indent()
-                ->subcompile($this->getNode('else'))
-                ->outdent()
-                ->write("}\n")
-            ;
+            $compiler->subcompile($this->getNode('else'));
```

Comment:
The goal of this PR is to get the line number of `{% else %}` above the PHP `if` condition.

Your suggestion would print the line number of the line after the `{% else %}`.

Reply:
You can see it in the example of the PR description.

Reply:
I might be missing something. Isn't it what you want?

```php
$_v0 = new \Twig\Runtime\LoopIterator([1, 2, 3, 4]);
yield from ($_v1 = function ($iterator, &$context, $blocks, $recurseFunc, $depth) {
    $macros = $this->macros;
    $parent = $context;
    foreach ($iterator as $context["_key"] => $context["key"]) {
        // line 3
        yield "    IF";
    }
    // line 5
    if (0 === $iterator->getIndex0()) {
        yield "    ELSE";
    }
    unset($context['_key'], $context['key']);
    $context = array_intersect_key($context, $parent) + $parent;
    yield from [];
})($_v0, $context, $blocks, $_v1, 0);
```

Reply:
I tried doing `->addDebugInfo($this->getNode('else'))` but that will print the wrong line (4 instead of 3). See:

```twig
1 {% for product in products %}
2    <h2>FOR</h2>
3 {% else %}
4    <p>No products found</p>
5 {% endfor %}
```

Before this PR it compiles to:
```php
// line 1
foreach ($context['_seq'] as $context["_key"] => $context["product"]) {
    // line 2
    yield "        <h2>FOR</h2>\n    ";
    $context['_iterated'] = \true;
}
if (!$context['_iterated']) {
    // line 4
    yield "        <p>No products found</p>\n    ";
}
```

With this PR it compiles to:
```php
// line 1
foreach ($context['_seq'] as $context["_key"] => $context["product"]) {
    // line 2
    yield "        <h2>FOR</h2>\n    ";
    $context['_iterated'] = \true;
}
// line 3
if (!$context['_iterated']) {
    // line 4
    yield "        <p>No products found</p>\n    ";
}
```

Because `if (!$context['_iterated']) {` points to `{% else %}` on line 3.

Reply:
Got it now. Looks like your PR makes sense.